src/dataset.js

```diff
@@ -554,6 +597,9 @@ Dataset.prototype.query = function(options, callback) {
  * Create a Table object.
  *
  * @param {string} id The ID of the table.
+ * @param {object} [metadata] Table metadata.
+ * @param {string} [metadata.location] The geographic location of the table.
+ *      Required except for US and EU.
```

Comment:
Can you remove the line about it being required? I'd suggest what we have in our discovery doc: The geographic location where the table resides. This value is inherited from the dataset.

Reply:
Also, can you make it clear that the table's location is read-only? This library should populate it from API responses, but callers should not populate it.

Reply:
Absolutely! I just pushed a commit with some updated verbiage